bucket-cache.go

```diff
@@ -147,7 +147,7 @@ func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, erro
 	urlValues.Set("location", "")
 
 	// Set get bucket location always as path style.
-	targetURL := c.endpointURL
+	targetURL := *c.endpointURL
```

Comment:
> Client.endpointURL can itself hold a value instead of pointer right? There is no use of it being a pointer
> It can avoid bugs like these

We need to re-do this and use string itself.